src/hidra/receiver/asapo_transfer.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import logging
+
+from hidra import Transfer
+from plugins.asapo_producer import AsapoWorker
+
+logger = logging.getLogger(__name__)
+
+
+class AsapoTransfer:
+    def __init__(self, endpoint, beamtime, token, n_threads, file_regex,
+                 default_data_source=None, timeout=5, beamline='auto',
+                 start_file_idx=1):
+
+        self.signal_host = "localhost"
+        self.targets = [["localhost", "50101", 1]]
+        self.asapo_worker = AsapoWorker(endpoint, beamtime, token, n_threads, file_regex,
+                                        default_data_source, timeout, beamline,
+                                        start_file_idx)
+
+        self.start_transfer()
+
+    def start_transfer(self):
+
+        query = Transfer("STREAM_METADATA", self.signal_host)
+        query.initiate(self.targets)
+        query.start()
+        self.run(query)
+        query.stop()
+
+    def run(self, query):
+        while True:
```

Comment:
Better add a `stop` method to stop the loop and then call it in a signal handler.